### DISCUSSION TITLE
Horizontal list with multiple rows

### DISCUSSION THREAD

**Discussion by Mohamed-kassim** [Shopify/flash-list#571] [General]
Any recommended way for implementing a horizontal list with multiple rows, it's easy to implement in Flipkart/RecyclerListView as we provide height and width to the layout handler
also I don't think  wrapping the list with scroll view like this solution, is the performant one
[https://stackoverflow.com/questions/45939823/react-native-horizontal-flatlist-with-multiple-rows](url)

**Comment by davebcn87** [comment]
Hey @Mohamed-kassim, thanks for you message. I converted this to a discussion as we don't consider this a bug. Can you send a screenshot of what you are trying to achieve?

**Reply by Mohamed-kassim** [reply]
Hey @davebcn87  you are right, this shouldn't be considered a bug
what I am trying to achieve is a horizontal list, but not with a single row
![photo_2022-08-24_16-28-02](https://user-images.githubusercontent.com/37451755/186445337-08563dfb-f56a-4c55-a955-02973b78b83e.jpg)
as you see here there are horizontal categories list with 2 rows, and horizontal products list with 3 rows
currently, we r using Flipkart/RecyclerListView, only by providing height for the item and giving the list parent container height
the list wraps items with the needed number of rows, the same for the products list
it's not required for me to calculate these rows dynamically I just need to implement the horizontal list with multiple rows.
if you checked the link for StackOverflow you will find old workarounds to get this in flat list.
and this is another image from there explaining the same requirement
![X3IMP](https://user-images.githubusercontent.com/37451755/186446496-dcf9d64d-7a69-4dec-9c12-5b95afa5d204.png)

**Comment by Mohamed-kassim** [comment]
@davebcn87 sorry for bothering, just a reminder

**Comment by naqvitalha** [comment]
@Mohamed-kassim Unfortunately, FlashList can only do one item per column when in horizontal mode. A quick workaround can be to wrap your entire column in a `View` and render it as a single item. Let me know if this helps.
